### PR TITLE
Updated to the latest StbSharp

### PR DESCRIPTION
**Pull request by rds1983**
Following up #6355, this PR adds StbSharp as submodule.
It's important to note that latest StbSharp is no longer monolithic, but split into different libs: https://github.com/StbSharp

Thus the PR adds two submodules which MonoGame uses: StbImageSharp and StbImageWriteSharp.
Their classes aren't exposed public, because of special STBSHARP_INTERNAL definition.
Also it's important to note that submodules of both projects reference special 'monogame' branch, that has  nothing but just those libs(i.e. no testing frameworks and no test data either).

**Update.** On the second thought, I've made it so StbImageSharp/StbImageWriteSharp 'master' branch contains only required stuff(moving non-required to branch 'develop'). Thus now MonoGame links 'master' branch of both libs.

### CODE REVIEW

File: MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj [hunk 1]
```diff
@@ -58,6 +58,9 @@ This package provides you with MonoGame Framework that uses OpenGL for rendering
     <Compile Include="Platform\Graphics\GraphicsAdapter.Legacy.cs" />
     <Compile Include="Platform\GamePlatform.Desktop.cs" />
     <Compile Include="Platform\GraphicsDeviceManager.SDL.cs" />
+    
+    <Compile Include="..\ThirdParty\StbImageSharp\src\StbImageSharp\**\*.cs" LinkBase="Utilities\StbImageSharp"/>
+    <Compile Include="..\ThirdParty\StbImageWriteSharp\src\StbImageWriteSharp\**\*.cs" LinkBase="Utilities\StbImageWriteSharp"/>
```
Comment (Jjagg) [inline MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj]: Oh neat, I did not know about LinkBase. We should use it for NVorbis stuff below too.

Comment (rds1983) [reply MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj]: Easy.

**Comment by Jjagg** [comment]
Building of docs is failing:

> [13:36:39][exec] SharpDoc(1,1): info:Generating documentation for namespace [StbImageSharp]
[13:36:45][exec] (1,1): error:Unexpected exception
[13:36:45][exec] (1,1): error:System.ArgumentException: Illegal characters in path.

I don't know what's up :/ It's dumb that it doesn't actually print the path or the illegal characters.

**Comment by rds1983** [comment]
How do you I run building docs?

**Comment by Jjagg** [comment]
Here's what the build bot does: 
```
<exec program="SharpDoc.exe" basedir="ThirdParty\Dependencies\SharpDoc" commandline="-config Documentation\config.xml" />
```

**Comment by rds1983** [comment]
Generation of documentation had been fixed.

**Comment by Jjagg** [comment]
Great, thanks @rds1983! :)

**Pull request merged**